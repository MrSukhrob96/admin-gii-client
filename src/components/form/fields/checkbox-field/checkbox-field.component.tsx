import { hoc } from '@utils/hoc';
import { useCheckboxFieldProps } from './checkbox-field.props';
import { CheckboxFieldContainer } from './checkbox-field.style';

/**
 * <CheckboxField />
 */
export const CheckboxField = hoc(useCheckboxFieldProps, () => (
  <CheckboxFieldContainer>
    {/* content */}
    {/* content */}
    {/* content */}
  </CheckboxFieldContainer>
));
