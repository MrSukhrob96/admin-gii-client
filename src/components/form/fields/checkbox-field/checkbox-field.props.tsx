
/**
 * <CheckboxField /> props
 */
export type CheckboxFieldProps = {

}

export const useCheckboxFieldProps = (_?: CheckboxFieldProps) => ({});

