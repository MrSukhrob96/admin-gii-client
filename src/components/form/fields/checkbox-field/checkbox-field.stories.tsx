import { ComponentStory, ComponentMeta } from '@storybook/react';
import { CheckboxField } from './checkbox-field.component';
import '@styles/index.css';

export default {
  title: 'Common/CheckboxField',
  component: CheckboxField,
} as ComponentMeta<typeof CheckboxField>

const Template: ComponentStory<typeof CheckboxField> = (args) => <CheckboxField {...args} />;

export const CheckboxFieldInitial = Template.bind({});
CheckboxFieldInitial.args = {
}