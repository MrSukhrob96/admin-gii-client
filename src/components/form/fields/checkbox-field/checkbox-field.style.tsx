import styled from 'styled-components'

export const CheckboxFieldContainer = styled.div`

`