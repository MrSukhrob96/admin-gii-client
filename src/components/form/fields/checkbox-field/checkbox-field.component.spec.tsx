import { customRender } from '@test-utils';
import { CheckboxField } from './checkbox-field.component';

describe('CheckboxField', () => {
  test('Should render CheckboxField component', async () => {
    const { asFragment } = customRender(
      <CheckboxField />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});