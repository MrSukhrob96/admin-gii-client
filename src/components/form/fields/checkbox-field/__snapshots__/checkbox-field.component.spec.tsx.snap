// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxField Should render CheckboxField component 1`] = `
<DocumentFragment>
  <div
    class=""
  />
</DocumentFragment>
`;
